### PR TITLE
Tinytest README - first draft

### PR DESCRIPTION
**Pull request by robfallows**
@tmeasday : PR as requested. Note I have retained the guide style (it's nice and friendly), but removed references that suggest this is a part of the guide.

Also, I've removed the "how do I test ..." sections. I no longer think they are appropriate in the context of a README, and especially as the official guide position will not be recommending Tinytest. There are resources around which go into this in some detail and links to those should probably be added.

I've added a reference to smithy:describe as you asked.

### CODE REVIEW

**Comment by stubailo** [comment]
This is awesome!

**Comment by stubailo** [comment]
I think this is much better than what we have now, and correct as far as I can see. Thank you!

**Comment by tmeasday** [comment]
Thanks so much @robfallows, this is really great

**Comment by robfallows** [comment]
@stubailo, @tmeasday : you're very welcome. Thanks for the opportunity to contribute :smile:

File: packages/tinytest/README.md [hunk 1]
```diff
@@ -1,26 +1,314 @@
...
+
+> ... Ideally, each test case is independent from the others. Substitutes such as method stubs, mock objects, fakes, and test harnesses can be used to assist testing a module in isolation.
+
+The object of a unit test is to prove functionality you have written. It is not to prove third-party or other package or library code which may form a part of your codebase. This may require that you head up your tests with *stubs, mocks/spies* or *fakes* which satisfy the demands of your code, but are non-functional (or have limited functionality). Alternatively, instead of placing the stubs in your test file, you could put them into a separate `stubs.js` file, which you `api.addFiles`, or even create and add a stubs package which you then `api.use`.
+
+From [martinfowler.com](http://www.martinfowler.com/articles/mocksArentStubs.html):
+
+> Fake objects actually have working implementations, but usually take some shortcut which makes them not suitable for production (an in memory database is a good example).
```
Comment (covall) [inline packages/tinytest/README.md]: Minimongo is a good example that the last example here is not correct or rather out of date.

**Pull request merged**